pkg/controller/controllerutil/controllerutil.go

```diff
@@ -180,6 +182,10 @@ const ( // They should complete the sentence "Deployment default/foo has been ..
 	OperationResultCreated OperationResult = "created"
 	// OperationResultUpdated means that an existing resource is updated
 	OperationResultUpdated OperationResult = "updated"
+	// OperationResultUpdatedStatus means that an existing resource and its status is updated
+	OperationResultUpdatedStatus OperationResult = "updatedStatus"
```

Comment:
These are currently only used for patch operations. The naming may be confusing since an update implies a full replace, while a patch does not. Consider `OperationResultPatched` + `OperationResultPatchedStatus`?

Reply:
I guess it doesn't really matter since we're diffing the object after mutation and deriving the patch based on that, though.

Reply:
Hi @alexeldeib,

> These are currently only used for patch operations. The naming may be confusing since an update sets the whole object, while a patch does not. Consider `OperationResultPatched `+ `OperationResultPatchedStatus`?

The reason for the current approach is to be consistent with the existing constants.

Reply:
what's the difference between `OperationResultUpdatedStatus` and `OperationResultUpdated`? does it make sense to expose this distinction in `CreateOrUpdate` as well? If not, does it make sense to remove one?

Reply:
> what's the difference between `OperationResultUpdatedStatus` and `OperationResultUpdated`? does it make sense to expose this distinction in `CreateOrUpdate` as well? If not, does it make sense to remove one?

Hi @alexeldeib,

I suppose if we do not create `Patched` variants, then there is no real distinction between `OperationResultUpdated` and `OperationResultUpdatedStatus`. Good catch. Here are my general thoughts on the matter:

* To me `Updated` means the result, not the method used to achieve it, another reason I did not create the variants
* It was important to me to have a result that indicated the status sub-resource was updated, but not the primary resource itself.

Beyond the above, I am really fine doing whatever here.

Reply:
my initial reaction was because of a race condition like this:

thread a - createOrUpdate
1. get object
2. mutate and extract patch
3. send patch to API server

thread b - racer
1. get obj
2. mutate 
3. send to apiserver

semantically I would want to tell the user their patch was successfully applied but I cannot guarantee that it was applied against the same object they provided. This is a slightly less strong guarantee than `OperationResultUpdated`, so I would have matched the HTTP methods and reached for `OperationResultPatched`. 

To me `OperationResultUpdatedStatus` means what I think `OperationResultUpdatedStatusOnly` is supposed to mean -- that a patch was applied but only the status was updated. Neither of them actually means an update operation was what was executed, vs the naming of `OperationResultCreated`, `OperationResultUpdated`

Reply:
Uhm what if both the object and the status were updated? Then we basically need to return two results or not?

Reply:
If both were updated, then `OperationResultUpdated` would imply both were updated. 

Again, I didn't change the names of the results to `Patched` because I was trying to preserve compatibility with the existing result values. I'm happy to adjust the names.

Reply:
But the fact that the `CreateOrPatch` tries to update the `Status` whereas `CreateOrUpdate` does not is just an implementation detail and not inherent to doing patch vs update or am I missing something?

May I ask for the reason you need to differentiate if the status resource was updated? I am asking because I believe that whatever we do here should also be added to `CreateOrUpdate` and I know ppl use the result sometimes to determine how to proceed. If we extend this we might break them which is why maybe returning a `[]result` may be better (but nor completely sure).